File: dbflow/src/main/java/com/raizlabs/android/dbflow/structure/BaseQueryModel.java
package com.raizlabs.android.dbflow.structure;

import androidx.annotation.NonNull;

import com.raizlabs.android.dbflow.annotation.QueryModel;
import com.raizlabs.android.dbflow.structure.database.DatabaseWrapper;

/**
 * Description: Provides a base class for objects that represent {@link QueryModel}.
 */
public class BaseQueryModel extends NoModificationModel {

    @Override
    public boolean exists() {
        throw new InvalidSqlViewOperationException("Query " + getClass().getName() +
                " does not exist as a table." + "It's a convenient representation of a complex SQLite query.");
    }

    @Override
    public boolean exists(@NonNull DatabaseWrapper databaseWrapper) {
        return exists();
    }

}

File: dbflow/src/main/java/com/raizlabs/android/dbflow/structure/BaseQueryModel.kt
package com.raizlabs.android.dbflow.structure

import com.raizlabs.android.dbflow.annotation.QueryModel
import com.raizlabs.android.dbflow.structure.database.DatabaseWrapper

/**
 * Description: Provides a base class for objects that represent [QueryModel].
 */
@Deprecated("No subclass needed. Use extension methods instead.")
class BaseQueryModel : NoModificationModel() {

    override fun exists(wrapper: DatabaseWrapper): Boolean {
        throw NoModificationModel.InvalidSqlViewOperationException("Query " + wrapper.javaClass.name +
                " does not exist as a table." + "It's a convenient representation of a complex SQLite query.")
    }
}
